### ISSUE TITLE
Google play rejection - anyone else?

### ISSUE THREAD

**Issue by nholmvio** [larryaasen/upgrader#188]
Hi -  I just got a Google play rejection for metadata call out as: 

We found an issue in the following area(s): 
 Full description (en_US): “[:mav: 2.1.11]”

I've been doing many releases with the description as is and using the upgrader package *with* success up until yesterday.  

Just posting this to see if anyone else is encountering an issue currently for Android that maybe the [mav] tag is throwing something off suddenly on google side?  I've removed the [mav] tag and attempting to resubmit for review, as i'm not sure why the description that has been fine for months is no longer ok and wondering if there is an issue with parsing this tag somehow at the end of the description?  Certainly realize this could be something unrelated...just trying to get a data point.

thanks

**Comment by nholmvio** [comment]
Just an update - after removing the [mav] tag....the release was approved on re-review with no other change...

So, i'm curious to hear as others do their Android releases going out the door around now!  
Thanks

**Comment by ahndwon** [comment]
Also happened to me.

**Comment by althurzard** [comment]
Same here

**Comment by nholmvio** [comment]
anyone try contacting Google yet on this?

**Comment by joymyr** [comment]
Just got the same rejection. And looking at the metadata policy https://support.google.com/googleplay/android-developer/answer/9898842/, it's clear that this tag doesn't comply. Anyway I created a ticket describing why we was using this tag, and asked them if they could suggest another way of adding the version to the description.

**Comment by Recker4244** [comment]
have any of you tried different packages? Does it do things differently? because this package doesn't seem to work unless I add that tag to my description. It returns the latest version as null

**Comment by nholmvio** [comment]
@joymyr  Thanks for the additional info and creating the ticket!  Any update back from them?

@Recker4244  I have not so far.  My understanding is that with the tag not there, it's just not enforcing any minimum version nor can we currently anymore on Android.  IOS appears to still be fine.  I'll have to confirm this in my next test cycle

@larryaasen   - Any thoughts on how to address for Android?

Thanks everybody

**Comment by larryaasen** [comment]
I'm interested in what @joymyr hears from Google about this, as the lack of Play Store features has forced us to look for hacky solutions like this.

@Recker4244 This package should work fine with or without the tag in the description. Perhaps you can open up a new [issue here](https://github.com/larryaasen/upgrader/issues/new/choose) and we can take a look at your problem.

I don't think #179 is related to this problem because that was a change with retrieving the Play Store details.

The format of the tag we are now using is like this: `[:mav: 1.2.3]`. Is there another parsable format that would be allowed by the Play Store?

**Comment by joymyr** [comment]
I haven't received any response from Google yet. I asked if another format would be accepted. But a human readable format would probably be accepted, as it wouldn't break the policy about non-descriptive information. It would be less optimal to parse though.

**Comment by nholmvio** [comment]
@joymyr @larryaasen   Larry - what's your thoughts on just a full text 'Minimum App Version x.x.x' at the bottom that is full spelled out to get Android back up?

**Comment by larryaasen** [comment]
I guess it is possible. The [current regex](https://github.com/larryaasen/upgrader/blob/6ae86349aa6629db0f9a5ebb67c5cd50c87eea54/lib/src/play_store_search_api.dart#L100) is:
```
String regExpSource = r"\[\:tagName\:[\s]*(?<version>[^\s]+)[\s]*\]";
```
We just need a new valid regex and we can try.

**Comment by ekerik220** [comment]
What does the minAppVersion field of Upgrader do? If we supply this do we not need the tag in the description?

**Comment by larryaasen** [comment]
```
minAppVersion: the minimum app version supported by this app. Earlier versions of this app will be forced to update to the current version. Defaults to null.
```
The store tag overrides any value supplied in the minAppVersion parameter.

**Comment by joymyr** [comment]
I updated the app description with this tag at the bottom, and it was accepted by Goggle Play:
> [Minimum supported app version: 1.1.0]

I also made a PR as a suggested change to parse that tag:
https://github.com/larryaasen/upgrader/pull/204

I also updated the tests, and it seems ok. But I'm not sure how Appcast works, or if it could mess up something there. Also, I'm not sure if it should be changed for App Store as well to match.

**Comment by polarby** [comment]
Any update on this proposed merge?

**Comment by larryaasen** [comment]
I just released a Prerelease: [4.7.0-alpha.1](https://pub.dev/packages/upgrader/versions/4.7.0-alpha.1), with a change to resolve this. Please test this out and let me know if this works for an Android Play Store submission.

**Comment by joymyr** [comment]
Great! I'll test this soon with my next app release.

**Comment by larryaasen** [comment]
Any feedback on this [4.7.0-alpha.1](https://pub.dev/packages/upgrader/versions/4.7.0-alpha.1) alpha release?

**Comment by joymyr** [comment]
I haven't released a new version of my app yet (will do that in a few days). But I have published an updated description with the new tag, and verified using a test-build with a lower version. It all seems great. I'm just not sure if they'll check the description more closely when updating the app, but I believe it will be fine.

**Comment by joymyr** [comment]
I did release an app using [4.7.0-alpha.1](https://pub.dev/packages/upgrader/versions/4.7.0-alpha.1) and with `[Minimum supported app version: 1.1.0]` at the bottom of the description. This works great.

**Comment by nholmvio** [comment]
@larryaasen @joymyr 
Larry Jon - Thanks everyone...appreciate the support on this.  I'll be releasing in a few weeks and will grab this as well.

Thanks again!
Nate

**Comment by larryaasen** [comment]
This was resolved with the [4.7.0](https://pub.dev/packages/upgrader/changelog#470) release.
